Entire gamelobby base refactor

CnCRAZER opened the pull request:
- Refactor `gamelobbybase.ini`
- Refactor skirmish and multiplayer lobbies
- Consolidated crate modifier dropdown next to crate checkbox
- Redesigned crate dropdown menu options
- Eliminate `GenericWindowLegacy`, there is no need to *spaghetify* different menu UI's with different ini files.

Disabled for now, pending beta spawner getting support for it:
- Add `Lock Game Speed` toggle
- Patch spawner to support new `Lock Game Speed` toggle

<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/4bac60d9-1257-4732-a259-dfa793d13bf3" />

<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/a03e7a5f-f7eb-42d0-b629-0ffd6252f324" />

CnCRAZER committed:
Revamp crate settings and update lobby UI layout

Removed Disabled.ini and renamed Super Increased.ini to Extreme.ini for crate options.

Added Minimal.ini for minimal crate settings.

Overhauled GameLobbyBase.ini and related lobby INIs to improve layout, control spacing, and option grouping.

Updated crate and AI modifier dropdowns, adjusted control positions, and improved theme constants for more consistent UI spacing.

Updated binary and updateexec to reflect these changes.

CnCRAZER committed:
Add btnSaveLoadGameOptions to MultiplayerGameLobby UI

Introduced the btnSaveLoadGameOptions button to MultiplayerGameLobby.ini with specific size and position settings. Also added commented-out position logic to GameLobbyBase.ini for the same button. Removed redundant PlayerOptionsPanel and GameOptionsPanel properties from MultiplayerGameLobby.ini to clean up configuration.

CnCRAZER committed:
Update MultiplayerGameLobby.ini

whoops

CnCRAZER committed:
Update ForcedOptions.md

CnCRAZER commented:
@SadPencil @MahBoiDeveloper 

Would you mind adjusting your translations for these changes?

Only thing to note:
Lock Game Speed is now a checkbox.
-> If enabled, gamespeed slider does not show up while ingame.
-> If disabled, gamespeed slider shows up as normal.

Crate Modifier:
-> Disabled is now Minimal. Maximum and minimum of 1 crate at all times.
-> Enabled is now Normal. Whatever map values are set inside the map are not changed. If none are set, it takes values from `spawner.xdp`
->Increased -> Minimum of 8 crates at all times
-> Extreme -> Minimum of 40 crates at all times

Blind Start:
Same as chkNoSpawnPreviews. Simply renamed to make it fit within the middle column

Dog Engi Immune:
Same as chkNoDogEngiEat: Simply renamed to make it fit within the right column

CnCRAZER committed:
Update ForcedOptions.md

SadPencil commented:
> @SadPencil @MahBoiDeveloper
> 
> Would you mind adjusting your translations for these changes?
> 
> Only thing to note: Lock Game Speed is now a checkbox. -> If enabled, gamespeed slider does not show up while ingame. -> If disabled, gamespeed slider shows up as normal.
> 
> Crate Modifier: -> Disabled is now Minimal. Maximum and minimum of 1 crate at all times. -> Enabled is now Normal. Whatever map values are set inside the map are not changed. If none are set, it takes values from `spawner.xdp` ->Increased -> Minimum of 8 crates at all times -> Extreme -> Minimum of 40 crates at all times
> 
> Blind Start: Same as chkNoSpawnPreviews. Simply renamed to make it fit within the middle column
> 
> Dog Engi Immune: Same as chkNoDogEngiEat: Simply renamed to make it fit within the right column

will do

MahBoiDeveloper commented:
> Would you mind adjusting your translations for these changes?
> 
> Only thing to note: Lock Game Speed is now a checkbox. -> If enabled, gamespeed slider does not show up while ingame. -> If disabled, gamespeed slider shows up as normal.
> 
> Crate Modifier: -> Disabled is now Minimal. Maximum and minimum of 1 crate at all times. -> Enabled is now Normal. Whatever map values are set inside the map are not changed. If none are set, it takes values from `spawner.xdp` ->Increased -> Minimum of 8 crates at all times -> Extreme -> Minimum of 40 crates at all times
> 
> Blind Start: Same as chkNoSpawnPreviews. Simply renamed to make it fit within the middle column
> 
> Dog Engi Immune: Same as chkNoDogEngiEat: Simply renamed to make it fit within the right column

Thank you for mention me to update text in translation. Would you kindly upload updated Translation.ini down below into this PR?

[Translation.zip](https://github.com/user-attachments/files/21989111/Translation.zip)

CnCRAZER commented:
> > Would you mind adjusting your translations for these changes?
> 
> > 
> 
> > Only thing to note: Lock Game Speed is now a checkbox. -> If enabled, gamespeed slider does not show up while ingame. -> If disabled, gamespeed slider shows up as normal.
> 
> > 
> 
> > Crate Modifier: -> Disabled is now Minimal. Maximum and minimum of 1 crate at all times. -> Enabled is now Normal. Whatever map values are set inside the map are not changed. If none are set, it takes values from `spawner.xdp` ->Increased -> Minimum of 8 crates at all times -> Extreme -> Minimum of 40 crates at all times
> 
> > 
> 
> > Blind Start: Same as chkNoSpawnPreviews. Simply renamed to make it fit within the middle column
> 
> > 
> 
> > Dog Engi Immune: Same as chkNoDogEngiEat: Simply renamed to make it fit within the right column
> 
> 
> 
> Thank you for mention me to update text in translation. Would you kindly upload updated Translation.ini down below into this PR?
> 
> 
> 
> [Translation.zip](https://github.com/user-attachments/files/21989111/Translation.zip)
> 
> 
> 
> 

Thanks, I'll upload it when I'm at my computer 💪

CnCRAZER committed:
Update UI layouts and window settings for lobbies

Refactored and unified window and control layouts across lobby and generic window INI files. Adjusted minimum ingame resolution, improved map search and chat message box positioning, and updated background and border settings for various windows. Deprecated or commented out unused controls and options, and switched LoadingScreen and MainLobby to use the new GenericWindow base. These changes improve consistency and flexibility in the UI.

CnCRAZER committed:
Update GameLobbyBase.ini

CnCRAZER committed:
Update ClientDefinitions.ini

MahBoiDeveloper commented on package/Resources/ClientDefinitions.ini:
```diff
@@ -25,10 +25,10 @@ MapEditorExePath=Map Editor\Map Editor.bat
+MinimumRenderWidth=1280;1024
+MinimumRenderHeight=768;600
+MaximumRenderWidth=1366;1600
+MaximumRenderHeight=800;900
```
Are you sure about these settings? Previous time changes make client a bit blurry (that was 1366x768 AFAIK).

brichardson1991 replied:
Just pulled this PR down and checked my skirmish screen:
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/dd277376-3b92-41c8-9219-74601128c6fd" />
settings:
<img width="1031" height="843" alt="image" src="https://github.com/user-attachments/assets/a4bac274-a9d9-43cc-81d2-eb0c59f0babf" />
same in multiplayer lobby
<img width="2560" height="1440" alt="image" src="https://github.com/user-attachments/assets/c5de832e-c1e6-4740-bee6-befa6ca30e33" />

CnCRAZER replied:
> Are you sure about these settings? Previous time changes make client a bit blurry (that was 1366x768 AFAIK).

This is 1366x768 currently.
I believe all my other refactor changes to make things a bit more scale-able have solved the blurriness, but other people confirming it will help.

<img width="1368" height="832" alt="image" src="https://github.com/user-attachments/assets/5a0c09b9-5609-4886-85fd-4969ada69134" />
<img width="1368" height="832" alt="image" src="https://github.com/user-attachments/assets/4f81e039-62d1-430b-ac2a-6d35358c0b5e" />

MahBoiDeveloper replied:
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/22e35a9f-7d19-4415-a8c2-0917660b5b45" />

Background need to edit too, or revert changes about resolution, because it creates black empty space on left and right corners.

CnCRAZER replied:
Ahhh nice catch. I'll look into that

CnCRAZER replied:
@MahBoiDeveloper This only seems to be an issue when in fullscreen mode. I suspect it's because I changed the min/max render sizes within client definitions.

Unless someone else can point out the fix for it(maybe resizing the loading.png file to fit this new format?) I don't see it as a huge issue as you see it for only 1 or 2 seconds

CnCRAZER replied:
ok @MahBoiDeveloper I believe as of my latest commit, this is fixed now

MahBoiDeveloper replied:
> ok @MahBoiDeveloper I believe as of my latest commit, this is fixed now

Yes, I confirm that, the issue has gone.

CnCRAZER committed:
Update lobby UI layout and add extra panel controls

Reordered and expanded render resolution options in ClientDefinitions.ini. In GameLobbyBase.ini, added new XNAExtraPanel controls for lobby sides and corners, adjusted control order, and improved layout for map preview and game mode labels. In GenericWindow.ini, reordered section definitions for clarity. These changes enhance lobby UI flexibility and support new decorative panel elements.

CnCRAZER committed:
Update Translation.ini

CnCRAZER committed:
Remove GenericWindowLegacy.ini and update updateexec

Deleted the GenericWindowLegacy.ini file and updated the updateexec script to reference its removal. This cleans up legacy resources and ensures the update process is consistent with the current resource set.

Any files using this legacy ini have been changed to reference GenericWindow.ini instead

CnCRAZER committed:
Update lobby UI layout and add extra panel elements

Adjusted various layout parameters in GameLobbyBase.ini for improved spacing and alignment. Added new extra panel controls for corners and sides in MainLobby.ini to support additional UI decorations. Made minor corrections to parentheses in MultiplayerGameLobby.ini and commented out some properties in GenericWindow.ini for flexibility.

CnCRAZER committed:
House Keeping

CCCP84 commented on package/INI/Game Options/Crates/Minimal.ini:
```diff
@@ -0,0 +1,3 @@
+[CrateRules]
+CrateMaximum=1
+CrateMinimum=1
```
The line CrateMaximum=1 would be enough. It works.
CrateMinimum=1 is not necessary to add.

CnCRAZER committed:
Update Minimal.ini

CCCP84 commented:
Remove 2.0 after YR Rebalance. Nobody needs this version number, or what does it mean..
The version number can be specified inside the file.
Although, I would delete the entire YR Rebalance together with the version number.. )

CnCRAZER committed:
Simplify YR Rebalance name

CnCRAZER commented:
> Remove 2.0 after YR Rebalance. Nobody needs this version number, or what does it mean.. The version number can be specified inside the file. Although, I would delete the entire YR Rebalance together with the version number.. )

I'm in favor of simplifying the name of the toggle to `YR Rebalance` vs `YR Rebalance 2.0`

CnCRAZER committed:
Update GameLobbyBase.ini

CnCRAZER committed:
Fix fullscreen loadscreen attempt 1

CnCRAZER committed:
loadscreen fullscreen fix attempt 2

CnCRAZER committed:
Update updateexec

CnCRAZER committed:
Merge branch 'develop' into feature/ui-fixes

SadPencil committed:
Update zh-Hans translation

SadPencil commented:
<img width="3204" height="1860" alt="图片" src="https://github.com/user-attachments/assets/a16aafc1-b4bd-43e9-b237-3a3886b0b1d0" />
finally, wide resolutions like 3200x1800 works without issues. Nicely done

SadPencil committed:
Translation adjustment following YR Rebalance 2.0

CnCRAZER committed:
restore spawner, revert gamespeed locking

CnCRAZER committed:
Merge branch 'feature/ui-fixes' of https://github.com/CnCNet/cncnet-yr-client-package into feature/ui-fixes

CnCRAZER commented:
> <img alt="图片" width="2000" height="1860" src="https://private-user-images.githubusercontent.com/11227602/485847031-a16aafc1-b4bd-43e9-b237-3a3886b0b1d0.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcwMTc5ODAsIm5iZiI6MTc1NzAxNzY4MCwicGF0aCI6Ii8xMTIyNzYwMi80ODU4NDcwMzEtYTE2YWFmYzEtYjRiZC00M2U5LWIyMzctM2EzODg2YjBiMWQwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA0VDIwMjgwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjYmQwMmUxY2VhZWQwMWFmYTZiNDk1OGI0YTVlY2M0Mjg4MDhlNDQwZjc3MjY1ODdjZmU5MTk3MTg1MWRlM2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ItJKGe3fLtZpbpZXN4L6j8wsFTIOepkRBhBWglJ0FS0"> finally, wide resolutions like 3200x1800 works without issues. Nicely done

Thank you for testing!!

CnCRAZER committed:
Enable 'Disable Game Speed' option in lobby

Uncomments and activates the 'Disable Game Speed' (Lock Game Speed) option in GameLobbyBase.ini, allowing users to disable in-game speed adjustment.

Also updates the gamemd-spawn.exe binary to support this feature.

brichardson1991 commented:
Hurry up and get this out of draft :P
